Mathlib/Geometry/RingedSpace/LocallyRingedSpace.lean

```diff
@@ -70,17 +70,32 @@ def 𝒪 : Sheaf CommRingCat X.toTopCat :=
 /-- A morphism of locally ringed spaces is a morphism of ringed spaces
  such that the morphisms induced on stalks are local ring homomorphisms. -/
 @[ext]
-structure Hom (X Y : LocallyRingedSpace.{u}) : Type _ where
-  /-- the underlying morphism between ringed spaces -/
-  val : X.toSheafedSpace ⟶ Y.toSheafedSpace
+structure Hom (X Y : LocallyRingedSpace.{u})
+    extends X.toPresheafedSpace.Hom Y.toPresheafedSpace : Type _ where
   /-- the underlying morphism induces a local ring homomorphism on stalks -/
-  prop : ∀ x, IsLocalRingHom (PresheafedSpace.stalkMap val x)
+  prop : ∀ x, IsLocalRingHom (PresheafedSpace.stalkMap toHom x)
+
+/-- A morphism of locally ringed spaces as a morphism of sheafed spaces. -/
+abbrev Hom.val {X Y : LocallyRingedSpace.{u}} (f : X.Hom Y) :
+  X.toSheafedSpace ⟶ Y.toSheafedSpace := f.1
+
+@[simp, nolint simpVarHead]
+lemma Hom.val_mk {X Y : LocallyRingedSpace.{u}}
+    (f : X.toPresheafedSpace.Hom Y.toPresheafedSpace) (hf) :
+  Hom.val ⟨f, hf⟩ = f := rfl
```

Comment:
I'm not sure what the `simpVarHead` issue is. The head symbol doesn't seem like a variable to me and it fires correctly. If this is still bad I can turn on this lemma locally.

Reply:
I made a few tests and I can confirm that the lemma triggers.